packages/ui/src/working-groups/components/OpeningsList/OpeningDetails.tsx

```diff
@@ -20,6 +23,36 @@ import { isOpeningOpen } from '@/working-groups/model/isOpeningOpen'
 
 export const OpeningDetails = ({ opening }: Props) => {
   const { showModal } = useModal()
+  const requiredStake = opening.stake.toNumber()
+  const { hasRequiredStake, transferableAccounts, accountsWithLockedFounds } = useHasRequiredStake(requiredStake)
+
+  const onApplyNowClick = useCallback(() => {
+    if (hasRequiredStake === true) {
+      showModal<ApplyForRoleModalCall>({ modal: 'ApplyForRoleModal', data: { opening } })
+    }
+    if (hasRequiredStake === false) {
+      if (transferableAccounts) {
+        showModal<MoveFundsModalCall>({
+          modal: 'MoveFundsModal',
+          data: { type: MoveFoundsModalType.TRANSFERABLE, accounts: transferableAccounts, requiredStake },
+        })
+      } else if (!transferableAccounts && accountsWithLockedFounds && Object.keys(accountsWithLockedFounds).length) {
+        showModal<MoveFundsModalCall>({
+          modal: 'MoveFundsModal',
+          data: {
+            type: MoveFoundsModalType.LOCKED_FOUNDS,
+            lockedFoundsAccounts: accountsWithLockedFounds,
+            requiredStake,
+          },
+        })
+      } else {
+        showModal<MoveFundsModalCall>({
+          modal: 'MoveFundsModal',
+          data: { type: MoveFoundsModalType.NO_FOUNDS, requiredStake },
+        })
+      }
+    }
+  }, [hasRequiredStake, transferableAccounts])
```

Comment:
I would move that to the `ApplyForRoleModal` this way we don't have to copy that check when invoking some modal. I don't recall if we have only one place with this button but there's a chance that we have at least two.

Reply:
The other thing is that the main modal, like ApplyForRole modal, could check only for `hasRequiredStake` and display the `<MoveFundsModal />`. The inner checks for which variant to display might be moved to the MoveFundsModall as well.

Reply:
Sorry, there was a slight misunderstanding. I understood that you wanted to rid the modal of all logic. I'll fix it 👍